dspace-api/src/main/java/org/dspace/content/authority/LocalMetadataLookup.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content.authority;
+
+import static org.dspace.content.authority.ChoiceAuthorityServiceImpl.CHOICES_INDEX_PREFIX;
+import static org.dspace.content.authority.ChoiceAuthorityServiceImpl.CHOICES_ORDER_PREFIX;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.FacetField;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.params.CommonParams;
+import org.apache.solr.common.params.FacetParams;
+import org.dspace.discovery.SearchUtils;
+import org.dspace.discovery.SolrSearchCore;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+public class LocalMetadataLookup implements ChoiceAuthority {
+
+    private static final Logger log = LogManager.getLogger(LocalMetadataLookup.class);
+
+    private String pluginInstanceName;
+
+    private String field;
+
+    private String order = "index";
+
+    String separator = SearchUtils.FILTER_SEPARATOR;
+
+    protected final ConfigurationService configurationService
+            = DSpaceServicesFactory.getInstance().getConfigurationService();
+
+    @Override
+    public Choices getMatches(String text, int start, int limit, String locale) {
+        SolrQuery solrQuery = new SolrQuery();
+
+        solrQuery.setQuery("*:*");
+        solrQuery.addFacetField(field);
+        solrQuery.add("f." + field + "." + FacetParams.FACET_LIMIT, String.valueOf(limit + 1));
+        solrQuery.setFacetPrefix(field, text.toLowerCase());
+        solrQuery.set(CommonParams.START, 0);
+        solrQuery.set(CommonParams.ROWS, 0);
+        solrQuery.add("f." + field + "." + FacetParams.FACET_SORT, order);
+        solrQuery.add("f." + field + "." + FacetParams.FACET_MINCOUNT, String.valueOf(1));
+        solrQuery.set(CommonParams.FQ, "search.resourcetype:Item AND latestVersion:true");
```

Comment:
Can you describe what this Solr query is searching for?  It would be good to describe it either in a code comment or in the methods JavaDocs, because it's not straightforward to me.